patroni/postgresql/__init__.py

```diff
@@ -255,7 +255,8 @@ def query(self, sql, *args, **kwargs):
             raise PostgresConnectionException(str(e))
 
     def data_directory_empty(self):
-        return not os.path.exists(self._data_dir) or os.listdir(self._data_dir) == []
+        return not os.path.exists(self._data_dir) or \
+                all(os.name != 'nt' and (n.startswith('.') or n == 'lost+found') for n in os.listdir(self._data_dir))
```

Comment:
1. Moving the `os.name != 'nt'` out of `all()` would make it slightly more readable . I imagine most readers expect only an iterable as an argument passed to `all`.
2. Windows apparently [also has hidden files](https://stackoverflow.com/a/14063074) but I guess it is out of scope

Reply:
1. If you move `os.name != 'nt'` out of `all()` result would be wrong.
2. src/port/pgcheckdir.c: pg_check_dir() runs exaclty the same check.